### PR TITLE
Add gnosis connector and sdk

### PR DESCRIPTION
**Pull request by samuveth**
Fixes https://github.com/snapshot-labs/snapshot/issues/813
Requires https://github.com/snapshot-labs/lock/pull/23

Changes:
- Update gnosis SDK and add the connector to UI
- Upgrade to new version of Lockjs
- Disable "Join" button when connected with Gnosis
- Throw error when trying to sign message with Gnosis

### CODE REVIEW

**Comment by samuveth** [comment]
@bonustrack what do you think, should we only show the Gnosis Safe connector in the connect wallet when inside a Gnosis Safe iframe?

**Comment by samuveth** [comment]
This too, I will wait for eip712 to get merge before finishing it off

File: package.json [hunk 1]
```diff
@@ -20,6 +20,8 @@
     "@ethersproject/strings": "^5.4.0",
     "@ethersproject/units": "^5.4.0",
     "@ethersproject/wallet": "^5.4.0",
+    "@gnosis.pm/safe-apps-provider": "^0.8.0",
+    "@gnosis.pm/safe-apps-sdk": "^4.3.0",
```
Comment (mktcode) [inline package.json:24]: Don't these belong to the lock repo? They are used [here](https://github.com/snapshot-labs/lock/blob/master/connectors/gnosis.ts) but not included in [package.json](https://github.com/snapshot-labs/lock/blob/master/package.json).

Comment (bonustrack) [reply package.json:24]: We aren't adding the dependencies on Lock.js directly, I think we do this to get more flexibility, but if we think its stable we could move all dependencies back to Lock

**Review by mktcode — APPROVED** [review]
Code looks fine.

**Comment by samuveth** [comment]
Continued here https://github.com/snapshot-labs/snapshot/pull/865

**Pull request closed without merging**